velox/dwio/common/CachedBufferedInput.h

```diff
@@ -13,48 +13,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 #pragma once
 
-#include <folly/Executor.h>
-
-#include "velox/common/caching/FileGroupStats.h"
-#include "velox/common/caching/ScanTracker.h"
-#include "velox/common/caching/SsdCache.h"
-#include "velox/dwio/common/BufferedInput.h"
-#include "velox/dwio/common/CacheInputStream.h"
-#include "velox/dwio/common/InputStream.h"
-#include "velox/dwio/common/IoStatistics.h"
+#include "velox/common/io/CachedBufferedInput.h"
 #include "velox/dwio/common/Options.h"
 
-DECLARE_int32(cache_load_quantum);
-
 namespace facebook::velox::dwio::common {
-
-struct CacheRequest {
-  CacheRequest(
-      cache::RawFileCacheKey _key,
-      uint64_t _size,
-      cache::TrackingId _trackingId)
-      : key(_key), size(_size), trackingId(_trackingId) {}
-
-  cache::RawFileCacheKey key;
-  uint64_t size;
-  cache::TrackingId trackingId;
-  cache::CachePin pin;
-  cache::SsdPin ssdPin;
-
-  bool processed{false};
-
-  // True if this should be coalesced into a CoalescedLoad with other
-  // nearby requests with a similar load probability. This is false
-  // for sparsely accessed large columns where hitting one piece
-  // should not load the adjacent pieces.
-  bool coalesces{true};
-  const SeekableInputStream* FOLLY_NONNULL stream;
-};
-
-class CachedBufferedInput : public BufferedInput {
+class CachedBufferedInput : public facebook::velox::io::CachedBufferedInput {
```

Comment:
We should also move this in the final version instead of creating unnecessary class inheritance

Reply:
There is not a low of usage of `CachedBufferedInput`, almost exclusively inside velox.  There are a lot of usage of `BufferedInput` though, those also need to be moved

Reply:
I would suggest you move them all in once (changes are massive but simple), instead of creating convoluted class hierarchy